File: day_03/solve_part1.rb
def main()
    file = File.open("input.txt")
    input = file.read
    pattern = /mul\([0-9]{1,3},[0-9]{1,3}\)/
    results = input.scan(pattern)
    total = 0
    results.each do |result|
        values = result.sub("mul(", "").sub(")", "").split(",").map(&:to_i)
        total += values[0] * values[1]
    end
    puts(total)
end

main()

File: day_19/solve_part1.py
import functools

with open("input.txt") as f:
    lines = f.readlines()

# Available towels
towels = list(sorted(map(lambda x: x.strip(), lines[0].strip().split(","))))
requests = list(map(lambda x: x.strip(), lines[2:]))

# Recursive function:
@functools.cache
def test_towels(prefix, request):
    results = []
    for towel in towels:
        if prefix + towel == request:
            return True
        elif request.startswith(prefix + towel):
            results.append(test_towels(prefix + towel, request))
    return any(results)

counter = 0
for pattern in requests:
    if test_towels('', pattern):
        counter += 1

print(counter)